src/main/java/nova/core/event/EventListenerList.java

```diff
@@ -0,0 +1,96 @@
+package nova.core.event;
+
+/**
+ * Implements a list of event listeners. This class is thread-safe and listeners
+ * can be added or removed concurrently, no external locking is ever needed.
+ * 
+ * @author Stan Hebben
+ * @param <T> event type
+ */
+public class EventListenerList<T> {
```

Comment:
I wanted to make it possible to supply a publish(event) method that would distribute an event to its listeners, so you can just create an EventListenerList, use the add/remove methods, and the publish method. This is to avoid having a for loop in every place an event listener list is used. It also automatically takes care of any concurrency / multi-threading issues that would come up with the use of listeners. You can just create a list and expect it to work.

Second, it's also an experiment to return the ListenerListHandle, which can be closed to unregister the listener easily (and efficiently). When implemented as a linked list, the handle can be the node object in the list - having no overhead while offering additional options. Closing the handle is much faster than removing an item from a list, and might pave the way towards some interesting low-lag implementations with listeners, such as blocks registering and unregistering on the fly to implement their interactions.